bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/config/impl/HttpCacheConfigImpl.java

```diff
@@ -0,0 +1,131 @@
+package com.adobe.acs.commons.httpcache.config.impl;
+
+import com.adobe.acs.commons.httpcache.config.HttpCacheConfig;
+import com.adobe.acs.commons.httpcache.store.HttpCacheStore;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.felix.scr.annotations.*;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Config class for http cache. Modelled as OSGi config factory. Every factory instance is tied to a request URI.
+ * Parameters such as selection of cache store, user groups, invalidation details shall also be configured.
+ */
+@Component(label = "ACS AEM Commons - HTTP Cache - Cache config",
+           description = "Config for request URI pattern that has to be cached. Each config is tied to a single " +
+                   "request URI pattern.",
+           configurationFactory = true,
+           metatype = true)
+@Service
+public class HttpCacheConfigImpl implements HttpCacheConfig {
+    private static final Logger log = LoggerFactory.getLogger(HttpCacheConfigImpl.class);
+
+    @Property(label = "Request URI pattern",
+              description = "Request URI pattern (REGEX) to be cached. Example - /content/mysite(.*).product-data" +
+                      ".json. Mandatory parameter.")
+    private static final String PROP_REQUEST_URI_PATTERN = "httpcache.config.requesturi.pattern";
+    private String requestUriPattern;
+
```

Comment:
It would be useful to have a Blacklist that is evaluated after the whitelist (Whitelist being "Request URI Pattern") .. This way it would be very easy to include `*.html` but exclude `*.nocache.*`